Client/WebAuthN/U2FExtensions.swift

```diff
@@ -121,83 +129,157 @@ class U2FExtensions: NSObject {
-                keySession.removeObserver(self, forKeyPath: #keyPath(YKFKeySession.u2fService.keyState))
+                keySession.removeObserver(self, forKeyPath: #keyPath(YKFAccessorySession.fido2Service.keyState))
+                keySession.removeObserver(self, forKeyPath: #keyPath(YKFAccessorySession.u2fService.keyState))
             }
         }
     }
+    
+    private var keyType: FIDO2ViewControllerKeyType = .unknown
+    
+    /**
+     Returns the service associated with the desired session.
+     */
+    private var keyFido2Service: YKFKeyFIDO2ServiceProtocol? {
+        get {
+            if keyType == .accessory {
+                return YubiKitManager.shared.accessorySession.fido2Service
+            } else {
+                if #available(iOS 13.0, *) {
+                    return YubiKitManager.shared.nfcSession.fido2Service
+                }
+            }
+            return nil
+        }
+    }
+    
+    private var keyU2fService: YKFKeyU2FServiceProtocol? {
+        get {
+            if keyType == .accessory {
+                return YubiKitManager.shared.accessorySession.u2fService
+            } else {
+                if #available(iOS 13.0, *) {
+                    return YubiKitManager.shared.nfcSession.u2fService
+                }
+            }
+            return nil
+        }
+    }
 
     init(tab: Tab) {
         self.tab = tab
         defer {
             observeSessionStateUpdates = true
-            observeKeyStateUpdates = true
+            observeServiceStateUpdates = true
         }
         
         super.init()
         
         let handleCancelButton: () -> PopupViewDismissType = {
-            let handle = self.currentHandle
-            self.cleanupPinVerificationPopup()
-            switch self.currentMessageType {
-            case .FIDO2Create:
-                self.sendFIDO2RegistrationError(handle: handle)
-            case .FIDO2Get:
-                self.sendFIDO2AuthenticationError(handle: handle)
-            case.FIDORegister:
-                self.sendFIDORegistrationError(handle: handle, requestId: self.requestId[handle] ?? -1, errorCode: U2FErrorCodes.other_error)
-            case .FIDOSign:
-                self.sendFIDOAuthenticationError(handle: handle, requestId: self.requestId[handle] ?? -1, errorCode: U2FErrorCodes.other_error)
-            case .FIDOLowLevel, .None:
-                break
-            }
+            self.handleUserCancel()
             return .flyDown
         }
-        
-        [touchKeyPopup, insertKeyPopup, pinVerificationPopup].forEach {
+
+        [touchKeyPopup, pinVerificationPopup].forEach {
             $0.addButton(title: Strings.keyCancel, tapped: handleCancelButton)
         }
         
         // Make sure the session is started
-        YubiKitManager.shared.keySession.startSession()
+        YubiKitManager.shared.accessorySession.startSession()
```

Comment:
Do we need to start `nfcSession` when on iOS 13+? I ask because the code before this shows `nfcSession` and it's guarded by an if-available statement and doesn't use `accessorySession`.

Reply:
We start the nfcsession here: https://github.com/brave/brave-ios/pull/1939/files#diff-de38284f9d7a27ae89ede4cf95b6aad6R730

NFC session unlike accessory session have a blocking UI. So we cannot enable it at start.